### PR TITLE
DLS-3499: Test endpoints for help to save reminders

### PR DESCRIPTION
**Pull request by LGoldfinch2**

### CODE REVIEW

File: conf/testOnlyDoNotUseInAppConf.routes [hunk 1]
```diff
@@ -10,4 +10,8 @@
 # Failing to follow this rule may result in test routes deployed in production.
 
 # Add all the application routes to the prod.routes file
+#
+GET     /help-to-save/test-only/populate-reminders/:noUsers/:emailPrefix/daysToReceive        uk.gov.hmrc.helptosavefrontend.controllers.test.TestController.populateHTSReminders(noUsers: Int, emailPrefix: String, day: List[Int])
+GET     /help-to-save-reminder/test-only/gethtsuser/:nino                                     uk.gov.hmrc.helptosavefrontend.controllers.test.TestController.getHTSUser(nino: String)
```
Comment (calinursaciuc) [inline conf/testOnlyDoNotUseInAppConf.routes:15]: I don't think this will be found in QA. You need to keep the base url as the frontend one.

Comment (calinursaciuc) [reply conf/testOnlyDoNotUseInAppConf.routes:15]: In theory both should start with /help-to-save/ if that's the frontend first part of the url.

File: app/uk/gov/hmrc/helptosavefrontend/controllers/AccessAccountController.scala [hunk 2]
```diff
@@ -44,7 +44,6 @@ class AccessAccountController @Inject() (
   errorHandler: ErrorHandler,
   maintenanceSchedule: MaintenanceSchedule,
   confirmCheckEligibility: confirm_check_eligibility,
-  errorTemplate: error_template
```
Comment (calinursaciuc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/AccessAccountController.scala:47]: maybe not delete this one

File: app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala [hunk 3]
```diff
@@ -59,7 +59,6 @@ class ReminderController @Inject() (
   reminderCancelConfirmation: reminder_cancel_confirmation,
   reminderDashboard: reminder_dashboard,
   applySavingsReminders: apply_savings_reminders,
-  notEligible: not_eligible
```
Comment (calinursaciuc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala:62]: maybe not delete this one

File: app/uk/gov/hmrc/helptosavefrontend/connectors/test/TestConnector.scala [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import play.api.Configuration
+import uk.gov.hmrc.helptosavefrontend.util.NINO
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
+import scala.annotation.tailrec
+import scala.concurrent.{ExecutionContext, Future}
+
+class TestConnector @Inject()(http: HttpClient, config: Configuration)(implicit ec: ExecutionContext) {
+
+  val conf: Config = config.underlying.get[Config]("microservice.services.help-to-save-reminder").value
+  val htsReminderUrl: String = "http://localhost:7008/help-to-save-reminder"
```
Comment (calinursaciuc) [inline app/uk/gov/hmrc/helptosavefrontend/connectors/test/TestConnector.scala:31]: will this work in QA? Make it configurable?

**Review by lutnos — CHANGES_REQUESTED** [review]
Not sure this is required, see Slack

**Pull request closed without merging**